include/deal.II/grid/grid_tools.h

```diff
@@ -756,6 +756,88 @@ namespace GridTools
   std::vector<typename MeshType::active_cell_iterator>
   compute_ghost_cell_halo_layer (const MeshType &mesh);
 
+  /**
+   * Extract and return the active cell within a @p skin_thickness
+   * around a subdomain (set of active cells) in the @p mesh
+   * (i.e. those that share a common set of vertices with the subdomain but
+   * are not a part of it). Here, the "subdomain" consists of exactly all of
+   * those cells for which the @p predicate returns @p true.
+   *
+   * The function first computes interior active, boundary cells of the
+   * subdomain that conform to the predicate.
+   * Using compute_bounding_box, a bounding box is
+   * computed for this subdomain and extended by @p skin_thickness. All the
+   * active cells that are outside the extended bounding box are ignored.
+   * The cells that are inside the extended bounding box are then checked for
+   * their proximity to the interior subdomain boundary cells.
+   *
+   * @tparam MeshType A type that satisfies the requirements of the
+   * @ref ConceptMeshType "MeshType concept".
+   * @param mesh A mesh (i.e. objects of type Triangulation, DoFHandler,
+   * or hp::DoFHandler).
+   * @param predicate A function  (or object of a type with an operator())
+   * defining the subdomain around which the halo layer is to be extracted. It
+   * is a function that takes in an active cell and returns a boolean.
+   * @param skin_thickness specifies the distance within
```

Comment:
@jppelteret 

> I'd be tempted to call the function compute_ghost_cell_layer_within_distance and the input argument layer_thickness.

that sounds good to me as well.

Probably you meant `compute_active_cell_layer_within_distance` because this function takes a general predicate?